### PR TITLE
Allow user to choose default ui camera

### LINKED ISSUE #11377: Allow user to choose the default ui camera
## What problem does this solve or what need does it fill?

There is an example of a problem that this can lead in #11237, where we cannot spawn a new Camera in the same window as primary window without possible problems appearing (Debug camera interpolating with Ui camera)

## What solution would you like?

- The creation of a marker Component DefaultUiCamera (or a similar name: This name is already in use and can lead to conflicts), so when selecting the camera with a root entity that doesn't have `TargetCamera`, the DefaultCamera param will give priority to this component. 

## What alternative(s) have you considered?

- The user, when creating a camera for debug purposes (Or another purposes that for now I don't know if exists), will have to add an TargetCamera for each root that exists (even though only one camera will load the Ui, the other Camera is created for another purposes) (This option Don't need the API Refactor)
- Return with the `UiCameraOptions`, an component that marks if the Camera should show ui or not (I think we don't need this anymore)

## Aditional Context

- For now, say that you have two cameras spawned for PrimaryWindow -> The actual API will not return a warn or something, it will almost randomly pick one of the cameras. The camera depends on the camera order, the line is as follow:
```rust

    cameras: Query<'w, 's, (Entity, &'static Camera)>,

///...
self.cameras
    .iter()
    .filter(|(_, c)| match c.target {
        RenderTarget::Window(WindowRef::Primary) => true,
        RenderTarget::Window(WindowRef::Entity(w)) => self.primary_window.get(w).is_ok(),
        _ => false,
    })
    .max_by_key(|(e, c)| (c.order, *e))
    .map(|(e, _)| e)
```
but using the order can conflict if the user is building an debug camera (That generally will be above others)

### PR DESCRIPTION
**Pull request by pablo-lua**
# Objective

- Resolves #11377 

## Solution

- Add marker component `IsDefaultUiCamera` that will be choosen first as the default camera.

If you want the IsDefaultUiCamera default camera to be in another window, thats now possible.
- `IsDefaultUiCamera` is expected to be within a single Camera, if that assertion fails, one PrimaryWindow Camera will be choosen.

---

### Added
- Added `IsDefaultUiCamera` marker component.

### CODE REVIEW

File: crates/bevy_ui/src/ui_node.rs [hunk 1]
```diff
@@ -1844,22 +1844,31 @@ impl TargetCamera {
     }
 }
 
+#[derive(Component)]
+/// Marker used to demarcate default cameras, they will have priority over `PrimaryWindow` cameras
```
Comment (alice-i-cecile) [inline crates/bevy_ui/src/ui_node.rs:1848]: It would be good to have a bit more context on why this might be useful in these docs.

Comment (matiqo15) [inline crates/bevy_ui/src/ui_node.rs:1848]:
```suggestion
/// Marker used to identify default cameras, they will have priority over `PrimaryWindow` cameras.
```

As a non-native English speaker, I had to google "demarcate" to understand this.

Comment (alice-i-cecile) [inline crates/bevy_ui/src/ui_node.rs:1848]:
```suggestion
/// Marker used to demarcate default cameras, they will have priority over the [`PrimaryWindow`] camera.
```

File: crates/bevy_ui/src/ui_node.rs [hunk 2]
```diff
@@ -1844,22 +1844,31 @@ impl TargetCamera {
...
+pub struct UiDefaultCamera;
```
Comment (matiqo15) [inline crates/bevy_ui/src/ui_node.rs:1849]:
```suggestion
pub struct DefaultUiCameraMarker;
```

File: crates/bevy_ui/src/ui_node.rs [hunk 3]
```diff
@@ -1844,22 +1844,31 @@ impl TargetCamera {
...
-        self.cameras
-            .iter()
-            .filter(|(_, c)| match c.target {
-                RenderTarget::Window(WindowRef::Primary) => true,
-                RenderTarget::Window(WindowRef::Entity(w)) => self.primary_window.get(w).is_ok(),
-                _ => false,
-            })
-            .max_by_key(|(e, c)| (c.order, *e))
-            .map(|(e, _)| e)
+        self.default_cameras.get_single().ok().or_else(|| {
```
Comment (alice-i-cecile) [inline crates/bevy_ui/src/ui_node.rs:1860]: We should emit a `warn_once` warning if there is more than one of these.

**Review by matiqo15 — APPROVED** [review]

File: crates/bevy_ui/src/ui_node.rs (hunk 1 above)
Comment (pablo-lua) [reply crates/bevy_ui/src/ui_node.rs]: Added the new context!

File: crates/bevy_ui/src/ui_node.rs [hunk 4]
```diff
@@ -1844,22 +1844,69 @@ impl TargetCamera {
...
```
Comment (doonv) [inline crates/bevy_ui/src/ui_node.rs:1894]:
```suggestion
        self.default_cameras.get_single().or_else(|_| {
```

File: crates/bevy_ui/src/ui_node.rs [hunk 5]
```diff
@@ -1844,22 +1844,69 @@ impl TargetCamera {
...
+///             },
+///             ..Default::default()
+///         },
+///         // We add the Marker here so all Ui will spawn in
+///         // another window if no TargetCamera is specified
+///         UiDefaultCameraMarker
+///     ));
+/// }
+/// ```
+pub struct UiDefaultCameraMarker;
```
Comment (doonv) [inline crates/bevy_ui/src/ui_node.rs:1883]: I'm not really a fan of the "Marker" suffix here, but at the same time not having the "Marker" suffix will make this components name very similar to the `DefaultUiCamera` `SystemParam`'s name.

File: crates/bevy_ui/src/ui_node.rs (hunk 4 above)
Comment (pablo-lua) [reply crates/bevy_ui/src/ui_node.rs:1894]: That would need to change the return from `Option<Entity>` to `Result<Entity, SomeError>`, I don't really want to change that thing in specific. DefaultUiCamera is used in ui_layout_tree as an explicit `Option`

File: crates/bevy_ui/src/ui_node.rs (hunk 5 above)
Comment (pablo-lua) [reply crates/bevy_ui/src/ui_node.rs:1883]: I don't really like that name too. We should for now think in another name for `DefaultUiCamera` or `UiDefaultCameraMarker`, but I don't have many ideas on this aspect 🤔

File: crates/bevy_ui/src/ui_node.rs (hunk 4 above)
Comment (doonv) [reply crates/bevy_ui/src/ui_node.rs:1894]: Ah my bad

File: crates/bevy_ui/src/ui_node.rs (hunk 5 above)
Comment (benfrankel) [reply crates/bevy_ui/src/ui_node.rs:1883]: `IsDefaultUiCamera` is an option.

Comment (pablo-lua) [reply crates/bevy_ui/src/ui_node.rs:1883]: `IsDefaultUiCamera` is a good option, for now, better than what we have.

Comment (alice-i-cecile) [reply crates/bevy_ui/src/ui_node.rs:1883]: Yep, let's move forward with that for 0.13.

**Review by alice-i-cecile — APPROVED** [review]

**Pull request merged**